src/test/javascript/portal/details/NcWmsPanelSpec.js

```diff
@@ -178,6 +178,85 @@ describe('Portal.details.NcWmsPanel', function() {
         });
     });
 
+    describe('_ncwmsParamsAsFilters', function() {
+
+        var expectedNcwmsParamsFilter;
+        var expectedDateFilterConfig;
+        var testDateFilter = {};
+
+        beforeEach(function() {
+            expectedNcwmsParamsFilter = {
+                isNcwmsParams: true/*,
+                hasValue: function() { return false; }*/
+            };
+
+            expectedDateFilterConfig = {
+                name: 'time',
+                value: {},
+                visualised: true
+            };
+
+            spyOn(Portal.filter, 'DateFilter').andReturn(testDateFilter);
+        });
+
+        it('updated start date', function() {
+
+            var testStartDate = moment();
+
+            var returnValue = ncwmsPanel._ncwmsParamsAsFilters(testStartDate, moment('invalid date'), null);
+
+            expectedNcwmsParamsFilter.dateRangeStart = testStartDate;
+            expectedDateFilterConfig.value.fromDate = testStartDate.toDate();
+
+            delete returnValue[1].hasValue; // Remove function so it isn't compared against
```

Comment:
I did this to get the tests to work. The hasValue() functions weren't passing the equality check (I'm guessing because the scope was different). This is a simple way to take them out of the equation.